processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -454,6 +454,14 @@ private DimensionSelector makeDimensionSelectorUndecorated(
                         );
                       }
 
+                      if (columnDesc.getCapabilities().getType() == ValueType.LONG) {
+                        return new LongWrappingDimensionSelector(makeLongColumnSelector(dimension), extractionFn);
```

Comment:
The only case these would get created is if a query has a dimension spec on a base numeric column but with an extractionFn applied, so exFn was never null